app.go

```diff
@@ -445,31 +450,101 @@ func ValidateApp(opts ...Option) error {
 	return app.Err()
 }
 
+func (app *App) constructCustomLogger(connectLogger func(logger fxevent.Logger)) error {
+	if err := app.container.Provide(app.fxlogConstructor); err != nil {
+		app.err = multierr.Append(app.err,
+			fmt.Errorf("could not construct custom logger via fx.WithLogger: %w", err))
+		connectLogger(fxlog.DefaultLogger(os.Stderr))
+
```

Comment:
nit:

 ```suggestion
```

Reply:
Here, I try and follow https://github.com/ssgreg/nlreturn. Do we disagree that this is a valid lint warning?

Reply:
That linter seems pretty arbitrary, TBH. I feel like whether a newline should be added in the function body or not is a matter of "paragraphs" of code with a shared purpose, and not something you can always lint for.

This return here is connected to the two statements above it, in that all three together are how we're choosing to handle that error. Like, how we handle that error is `(record, connect, return)`, not `(record, connect)` and an independent `return`. So to me the `return` is part of the paragraph containing the other two statements.

Reply:
ok, sounds good. i'll disable it from list that i use.